d2cli/watch.go

```diff
@@ -442,13 +510,18 @@ func (w *watcher) handleRoot(hw http.ResponseWriter, r *http.Request) {
 </body>
 </html>`, filepath.Base(w.outputPath), w.devMode)
 
+	w.boardpathMu.Lock()
 	// if path is "/x.svg", we just want "x"
 	boardPath := strings.TrimPrefix(r.URL.Path, "/")
 	if idx := strings.LastIndexByte(boardPath, '.'); idx != -1 {
 		boardPath = boardPath[:idx]
 	}
+	recompile := false
 	if boardPath != w.boardPath {
 		w.boardPath = boardPath
```

Comment:
Shouldn't there be a `recompile = true` here? As it is now, if you navigate to a layer (e.g. `/layers/x`), modify the root layer and then navigate back to `/`, you will see an outdated version.

Reply:
yeah oops thank you!